user/conditions-v0.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Conditions (v0, deprecated)
+layout: en
+---
+
+> This page documents the behavior of conditions `v0` (beta), which is
+> currently being deprecated. Please opt in to the new version `v1` using
+> `conditions: v1` in your `.travis.yml` file. The documentation for `v1`
+> can be found [here](/user/conditions-v1).
+
+Conditions can be used to filter out, and reject builds, stages, and jobs by
+specifying conditions in your build configuration (your `.travis.yml` file).
+See [Conditional Builds, Stages, and Jobs](/user/conditional-builds-stages-jobs)
+for details.
+
+## Specifying conditions
+
+The condition can be specified using a boolean language as follows:
+
+```
+(NOT [term] OR [term]) AND [term]
+```
+
+A term is defined as:
+
+```
+[left-hand-side] [operator] [right-hand-side]
+```
+
+All keywords (such as `AND`, `OR`, `NOT`, `IN`, `IS`, attributes, and
+functions) are case-insensitive.
+
+### Left hand side
+
+The left hand side part can either be a known attribute or a function call.
+
+Known attributes are:
+
+* `type` (the current event type, known event types are: `push`, `pull_request`, `api`, `cron`)
+* `repo` (the current repository slug `owner_name/name`)
+* `branch` (the current branch name; for pull requests: the base branch name)
+* `tag` (the current tag name)
+* `sender` (the event sender's login name)
+* `fork` (`true` or `false` depending if the repository is a fork)
+* `head_repo` (for pull requests: the head repository slug `owner_name/name`)
+* `head_branch` (for pull requests: the head repository branch name)
+
+Known functions are:
+
+* `env(FOO)` (the value of the environment variable `FOO`)
+
+The function `env` currently only supports environment variables that are given
+in your build configuration (e.g. on `env` or `env.global`), not environment
+variables specified in your repository settings.
+
+### Right hand side
+
+It is currently not possible to compare function calls. This means that if you
+try to evaluate something similar to:
+
+`env(PRIOR_VERSION) != env(RELEASE_VERSION)`
+
+where `PRIOR_VERSION` and `RELEASE_VERSION` are environment variables defined
+elsewhere in .travis.yml, the condition will be evaluated as true, even when it
```

Comment:
> the condition will always be evaluated as true, even..

I think that helps a bit for emphasis?

Reply:
@acnagy all of the text on `conditions-v0` is what is currently in production on https://docs.travis-ci.com/user/conditional-builds-stages-jobs/ ... i've just split this out to a separate page so we can have two version specific pages.

i'd rather not change this (and i wouldn't feel comfortable changing the workaround text as that was added by someone else later?)

Reply:
Hm, I think I wasn't clear with my comment - re: the workaround text - I was more just picking at defining a "work-around" in the official docs. If it's what we recommend in this case, it's what we recommend in this case. Instead of this:
> A work-around is to create a script which expresses the conditional test in reverse, ..

I'd recommend this:
> The best way to handle the behavior that comparison of function calls always evaluates to `true` is to create a script which expresses the conditional test in reverse, ...

Reply:
this all is to say, I don't particularly want to change the body of the text, just change the use of the word "work-around", to nod instead to the fact that this is how the official docs recommend handling this potential gotcha

Reply:
regarding where the doc came from (split from v0, and edited) -- good to know! that makes sense 👍

Reply:
subbed in my replacement for the word for "work-around", upgrading it to "the best way". Separately, the issue for `travis_terminate` is here: https://github.com/travis-ci/docs-travis-ci-com/issues/1899